### DISCUSSION TITLE
GraphQL audit queries return partial data

### DISCUSSION THREAD

**Discussion by ilia-cy** [community/community#24600] [Apps, API and Webhooks]
<p>Hello,</p>
<p>I’m trying to use GraphQL API to monitor audit logs in my organization.</p>
<p>For most of the queries it works great, but there are some actions that appear in the organization’s audit log page and are not retreivied via GraphQL API.</p>
<p>For example, all <em>protected_branch</em> and <em>hook</em> related actions can be seen via the audit log organization page, but retreive no results when searching for them using the GraphQL API.</p>
<p>They return nothing when using a <em>query:“action:protected_branch”</em> or when simply trying to retreive all entries without any filter whatsoever.</p>
<p>Is there a list of specific audit entries that are not accessible via GraphQL?<br>
Is there a plan of supporting them?</p>
<p>Thanks</p>

**Comment by ilia-cy** [comment, marked as answer]
<p>For anyone that is interested, I am attaching the answer from a Github support:</p>
<p><em>As you’ve found, the <a href="https://help.github.com/github/setting-up-and-managing-organizations-and-teams/reviewing-the-audit-log-for-your-organization#searching-the-audit-log" rel="nofollow noopener">audit log search query</a> action:protected_branch is not yet supported by our GraphQL API. The <a href="https://developer.github.com/v4/interface/auditentry/" rel="nofollow noopener">GraphQL AuditEntry interface reference</a> lists the currently supported audit log entry types.</em></p>
<p><em>Although this search query is likely to be supported by our GraphQL API at some point, I don’t have firm details at this time. If it’s added, there’ll be an announcement in our <a href="https://developer.github.com/changes/" rel="nofollow noopener">developer blog</a>, so it’s worth keeping an eye on that.</em></p>
<p>After specifiying that this is an important functionality, I also received the following reply:</p>
<p><em>I agree - this functionality would be extremely useful. As such, I’ve opened an internal feature request to add support for all audit log search queries to our GraphQL API.</em></p>
<p><em>Let me know if I can be of further help!</em></p>

**Comment by psalaberria002** [comment]
<p>Branch protection audit logs are among the most critical logs. It would be great if they were added to the GraphQL API.</p>
<p>Has there been any progress on this?</p>
<p>Thank you.</p>

**Comment by ilia-cy** [comment]
<p>I’ve replied to the email thread in which the support updated me that it was added as an internal feature request.</p>
<p>I’ve asked what is the status/ETA of the feature.</p>
<p>I’ll update when they will get back to me.</p>

**Comment by sahmeder** [comment]
<p>Hey I was wondering if you ever got a response back on status/ETA?</p>

**Comment by ilia-cy** [comment]
<p><a class="mention" href="/sahmeder">@sahmeder</a><br>
I did not receive any status/ETA.</p>

**Comment by sahmeder** [comment]
<p>Was told the same thing as well. Looks like we’ll just have to wait and see :slight_smile:</p>

**Comment by Dithilli** [comment]
I'm so desperately hoping there's an update to this. Any news?
